### ISSUE TITLE
Not obvious behavior in new build [e6bc949]

### ISSUE THREAD

**Issue by ermo4enkov** [MetaMask/metamask-extension#5656]
In the previous stable build, I have such flow:
1) Checking metamask installed
2) Checking user login
3) If it all ok, a user can work with metamask

Checking user login works that way:

```
const acc = web3.eth.getAccounts();
      acc.then(acc => {
        acc.length > 0 ? this.setAccount(acc.join('')) : false;
});
```

In a new build, I can't get user's accounts before the user not approved work with metamask. It looks fine, **but web3.eth.getAccounts()** without approved return [] and **window.ethereum.enable()** also return []; 

Do you see it looks like not obvious at all?
How can I check user logged in or he/she don't want to approve work with metamask?
Maybe it will work better if I can get a specific error message? For example "user doesn't approve".

**Browser details (please complete the following information):**
 - OS: macOS Mojave
 - Browser chrome
 - MetaMask Version [e6bc949]
 - Beta UI

**Additional context**
Here you can see when user doesn't logged in behavior
<img width="591" alt="screen shot 2018-11-02 at 12 55 36 pm" src="https://user-images.githubusercontent.com/13857128/47911624-a76df380-de9e-11e8-93dc-11813a8d8888.png">

**Comment by lovvtide** [comment]
I'm trying to reproduce this, but I'm not seeing the option to enable privacy mode in 4.17.0. Am I missing something?

**Comment by ermo4enkov** [comment]
Hello @lovvtide. I wrote about the latest build.

You have to install metamask latest build [e6bc949] manually, not from Chrome Store.

Here is the guide: https://medium.com/metamask/https-medium-com-metamask-breaking-change-injecting-web3-7722797916a8

Here is the latest build: https://github.com/MetaMask/metamask-extension/pull/4703#issuecomment-434615101

**Comment by JaceHensley** [comment]
I think if `getAccounts()` or `enable()` return an empty list, that means that it's locked. I just tried it out and that's what it seems like to me. Which is the same behavior pre-EIP-1102. There is also `window.ethereum._metamask.isUnlocked()` another helper function added in that PR

**Comment by ermo4enkov** [comment]
As I remember `window.ethereum._metamask.isUnlocked()` returns `true` even if a user doesn't log in. This method returns the cached value.

**Comment by JaceHensley** [comment]
Hmm I tried `inUnlocked()` and it resolved to `false` when I was logged out

**Comment by ermo4enkov** [comment]
Hello @JaceHensley, you right. I've checked again.

So are you agree with me it would be much more obvious if web3 will return a specific error?

**Comment by JaceHensley** [comment]
I think that would be a breaking change, since the current behavior is to return an empty list when the wallet is locked. And that may be how other dapp browsers do it

**Comment by bitpshr** [comment]
Hi @ermo4enkov and @JaceHensley, thanks very much for filing this issue. This was an interesting edge case: if the user explicitly locked MetaMask via the "logout" UI (as opposed to quitting and reopening the browser), we would return an empty array the next time `ethereum.enable()` is called instead of showing the approval popup so the user can log in as expected. This has been fixed as of 078f314c8c712d1c0743ddb1355a2c75a867c74b and the associated [builds](https://github.com/MetaMask/metamask-extension/pull/4703#issuecomment-435690892).

To summarize, both quitting the browser and locking MetaMask using the "logout" UI now behave identically: these actions both reset `ethereum.isUnlocked()` so it returns `true`, and calling `ethereum.enable()` again will show the approval popup so the user can log in. Apologies for the last-minute bugfix, and thanks again for identifying this discrepancy.

**Comment by ermo4enkov** [comment]
Hello @bitpshr does it mean I can close the issue?

**Comment by bitpshr** [comment]
Hi @ermo4enkov. Yes, feel free to close this issue if everything is working as expected for you with the builds at https://github.com/MetaMask/metamask-extension/pull/4703#issuecomment-435690892.

**Comment by ermo4enkov** [comment]
Thanks a lot! Have a nice day!
